### DISCUSSION TITLE
Composite-id class must implement Serializable: io.digiwave.entities.StoreEntity

### DISCUSSION THREAD

**Discussion by softshipper** [quarkusio/quarkus#26897] [Q&A]
Hi all,

I try to use Hibernate in Quarkus with Kotlin.   
The entity definition looks as follows:

    @Entity
    open class StoreEntity(
    ) : PanacheEntity() {
        @EmbeddedId
        lateinit var userId: UserId
        lateinit var password: String
        var createdAt: Long = 0
    }

and the composite key:

    @Embeddable
    open class UserId(
    ) : Serializable {
        lateinit var id: String
    }

Unfortunately the compiler complains:

    Caused by: org.hibernate.MappingException: Composite-id class must implement Serializable: io.digiwave.entities.StoreEntity
            at org.hibernate.mapping.RootClass.checkCompositeIdentifier(RootClass.java:293)
            at org.hibernate.mapping.RootClass.validate(RootClass.java:276)
            at org.hibernate.boot.internal.MetadataImpl.validate(MetadataImpl.java:359)
            at io.quarkus.hibernate.orm.runtime.recording.PrevalidatedQuarkusMetadata.validateAndWrap(PrevalidatedQuarkusMetadata.java:58)

As you can see the `Serializable` interface is implemented in the composite keys. 
What am I doing wrong? 

Best regards

[a comment by an automated account is not shown]

**Reply by evanchooly** [reply]
Make sure it's Java's `Serializable` and not Kotlin's.

**Reply by softshipper** [reply]
HI @evanchooly 
First of all thanks for your answer. I tried it out as follows:

UserId.kt
```
import java.io.Serializable
import javax.persistence.Embeddable

@Embeddable
open class UserId : Serializable {
    lateinit var id: String
}
```
StoreEntity.kt
```
import io.quarkus.hibernate.reactive.panache.PanacheEntity
import javax.persistence.EmbeddedId
import javax.persistence.Entity

@Entity
open class StoreEntity(
) : PanacheEntity() {
    @EmbeddedId
    lateinit var userId: UserId
    lateinit var password: String
    var createdAt: Long = 0
}
```

Unfortunately the same error message as I mentioned above.

You can find the example here
[kihibernate.zip](https://github.com/quarkusio/quarkus/files/9173269/kihibernate.zip)
